src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,107 @@
+const { expenseSchema } = require('../shemas');
+const { dynamicSchema } = require('../utils');
+const { expensesService, usersService } = require('../services');
+
+const getAll = async (req, res) => {
+  const query = req.query;
+
+  const expenses = await expensesService.getAll(query);
+
+  res.json(expenses);
+};
+
+const getOne = async (req, res) => {
+  const { id } = req.params;
+
+  const expense = await expensesService.getById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.json(expense);
+};
+
+const create = async (req, res) => {
+  const { error, value } = expenseSchema.validate(req.body, {
+    allowUnknown: true,
+  });
+
+  if (error) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const isUserExist = !!(await usersService.getById(value.userId));
+
+  if (!isUserExist) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const user = await expensesService.create(value);
```

Comment:
The variable `user` is used to store the result of `expensesService.create(value)`, but it would be more appropriate to name it `expense` to reflect the entity being created.